### PR TITLE
Working Hg38 version

### PR DESCRIPTION
**Pull request by NagaComBio**
This is the version with hg38 and hg19 working Beagle's phasing. Both reference genome works from this version and they produced identical CNVs.

This version will eventually be merged into the master. Currently, it is `naga:hg38 -> ACEseq:hg38 -> ACEseq:master` 
@vinjana We could also go for a new pull request directly to the `naga:hg38 -> ACEseq:master`, this will reduce some double review work.

### CODE REVIEW

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (vinjana) [inline resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh:23]: Did you try `${CNV_MPILEUP_OPTS[@]}"? It expands to the elements of a bash array and quotes each. This would also allow for unquoted elements that include whitespaces.

File: resources/configurationFiles/analysisCopyNumberEstimation.xml
Comment (vinjana) [inline resources/configurationFiles/analysisCopyNumberEstimation.xml:124]: Why is this quoted? Is it not required anymore at all? Does it mess things up? Can this be completely deleted?

**Review by vinjana — CHANGES_REQUESTED** [review]
Only minor comments.

The only thing I really would like to see, is an update of the changelogs an at least the information about a new/changed file (expected filename, short description of content and format).

File: resources/configurationFiles/analysisCopyNumberEstimation.xml
Comment (GWarsow) [reply resources/configurationFiles/analysisCopyNumberEstimation.xml:124]: these files are needed for noControl cases. we would have to define hg38 samples to be used as male/female genericControl samples. We still will need the *FAKE_CONTROL_PRE cvalues.

Comment (NagaComBio) [reply resources/configurationFiles/analysisCopyNumberEstimation.xml:124]: This was a TODO from Tom's version. Yes, it need only in no-control cases, and normal workflow runs through without this variable. @GWarsow does a leftover to hg38 coordinates works here?

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (NagaComBio) [reply resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh:23]: `$SAMTOOLS_BINARY mpileup "${CNV_MPILEUP_OPTS[@]}" -u -f` translates into `samtools mpileup '-A -R -B -Q 0 -q 1 -I ' -u -f /icgc/... ` The `'` is recognized as a value and generates wrong parameters as input.

**Review by vinjana — COMMENTED** [review]
Seems good to me, but it would be an overstatement to say I understood all the implications of the changes. Best @GWarsow provides the final approval.

**Comment by isidroc** [comment]
Hello @vinjana, any plans to accept the pull request of make an hg38-compatible version available? That would be great. Thanks a lot,
Isidro

**Review by vinjana — COMMENTED** [review]
O.k., all these changes will be on master after the merge, and master, anyway, has been changed previously (such as the switch to Beagle), such that backwards compatibility to the last major version 5.0.1 (with respect to the produced results) will not be given for hg37.

Like I said before, I am fine with the changes, but as @GWarsow is the current maintainer of the workflow and better understands what he is using it for, he should also give the final approval.

**Comment by GWarsow** [comment]
@NagaComBio Could you please provide information on how both potentialArtifacts files have been generated? Which input/assumptions/methods have been made/used here?

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (GWarsow) [inline resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh:23]: why did you remove the quotes here and in line 28?

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (GWarsow) [inline resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh:43]: could you please put this (or at least '9g') to the config so it will become easily adustable from outside? Same please for nthreads

**Comment by NagaComBio** [comment]
@GWarsow These are liftover of the artifact coordinates from hg19. The plan is to generate a new one once we have enough samples processed with the hg38 reference.

**Review by GWarsow — CHANGES_REQUESTED** [review]

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (NagaComBio) [reply resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh:23]: The translated shell script still contained the quotes, this created the issue during `mpileup` calls.

Comment (GWarsow) [reply resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh:23]: which value of CNV_MPILEUP_OPTS  caused issues for mpileup using quotes and no issue without quotes?

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (NagaComBio) [reply resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh]: @GWarsow Here is the previous comment about `CNV_MPILEUP_OPTS` issue.

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (NagaComBio) [reply resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh]: This is updated. But not tested yet. My job queues are currently full.

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (GWarsow) [reply resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh:23]: Thank you for pointing me to
https://github.com/DKFZ-ODCF/ACEseqWorkflow/pull/18#discussion_r576230951
Now I see what you meant.

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (GWarsow) [reply resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh]: thank you for updating this. could you please test it before pulling?

**Review by GWarsow — APPROVED** [review]

File: resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh
Comment (NagaComBio) [reply resources/analysisTools/copyNumberEstimationWorkflow/phasing.sh]: Tested, we could merge this now. @vinjana

**Review by vinjana — APPROVED** [review]

**Pull request merged**
